packages/react-component-library/src/components/Timeline/context/reducer.ts

```diff
@@ -103,24 +101,19 @@ export function reducer(
   action: TimelineAction
 ): TimelineState | never {
   switch (action.type) {
-    case TIMELINE_ACTIONS.SET_DATE:
-      return {
-        ...state,
-        today: action.date
-      }
     case TIMELINE_ACTIONS.GET_NEXT:
       return {
         ...state,
-        ...getMonths(addMonths(state.startDate, state.options.range), state),
-        weeks: getWeeks(addMonths(state.startDate, state.options.range), state),
-        days: getDays(addMonths(state.startDate, state.options.range), state)
+        ...getMonths(addMonths(state.startDate, state.options.rangeInMonths), state.options.rangeInMonths),
```

Comment:
Maybe we could destructure rangeInMonths to make these lines a bit more succinct.